tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -0,0 +1,863 @@
+                'FILEBEAT_ROOT_CA': '/etc/pki/filebeat/root-ca.pem',
+                'FILEBEAT_CERTIFICATE': f"/etc/pki/filebeat/node-{manager_index}.pem",
+                'FILEBEAT_KEY': f"/etc/pki/filebeat/node-{manager_index}-key.pem"
+            }
+            configuration_template_str = str(configuration_template)
+
+            for key, value in configuration_variables.items():
+                configuration_template_str = configuration_template_str.replace(key, value)
+
+            configurations[host] = ast.literal_eval(configuration_template_str)
+
+    return configurations
+
+
+@pytest.fixture(scope='module')
+def setup_vulnerability_tests(host_manager: HostManager) -> Generator:
+    """Setup the vulnerability tests environment
+
+    Args:
+        host_manager (HostManager): An instance of the HostManager class.
+    """
+    logger.error("Init setup of environment")
+
+    # Configure managers and agents
+    logger.error("Getting backup of current configurations")
+    hosts_configuration_backup = backup_configurations(host_manager)
+    logger.error("Configuring environment")
+    configure_environment(host_manager, load_vulnerability_detector_configurations(host_manager))
+
+    # Truncate alerts and logs of managers and agents
+    logger.error("Truncate managers and agents logs")
+    truncate_remote_host_group_files(host_manager, 'all', 'logs')
+
+    # Restart managers and stop agents
+    logger.error("Stopping agents")
+    host_manager.control_environment('stop', ['agent'])
+    logger.error("Restarting managers")
+    host_manager.control_environment('restart', ['manager'])
+
+    utc_now_timestamp = datetime.datetime.utcnow()
+
+    # Format the date and time as per the given format
+    test_timestamp = utc_now_timestamp.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # Wait until VD is updated
+    logger.error("Wait until Vulnerability Detector has update all the feeds")
+    wait_until_vd_is_updated(host_manager)
+
+    # Truncate alerts and logs of managers and agents
+    logger.error("Truncate managers and agents logs")
+    truncate_remote_host_group_files(host_manager, 'all', 'logs')
+
+    # Re-Register agents: https://github.com/wazuh/wazuh/issues/21185
+    logger.error("Removing agents")
+    host_manager.remove_agents()
+
+    # Wait until agents are registered again
+    time.sleep(15)
+
+    # Start agents
+    host_manager.control_environment('start', ['agent'])
+
+    yield test_timestamp
+
+    # Truncate alerts and logs of managers and agents
+    logger.error("Truncate managers and agents logs")
+    truncate_remote_host_group_files(host_manager, 'all', 'logs')
+
+    logger.error("Restoring original configuration")
+    restore_configuration(host_manager, hosts_configuration_backup)
+
+    logger.error("Restarting environment")
+    host_manager.control_environment('restart', ['agent'])
+    host_manager.control_environment('restart', ['manager'])
```

Comment:
All this should be more granular, and the test should only provide the variables to be configured.

Reply:
Agree. However, these setup include already encapsulated tasks: restart services in remote hosts, truncate remote logs, waiting expected time. Encapsulate this setup will be challenging and I suggest to go back to this issue in the future